File: 品味Java的21種設計模式/dp_cpp/state/example8/AuditOverState.cpp
#include "AuditOverState.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace state
			{
				namespace example8
				{

					void AuditOverState::doWork(StateMachine *request)
					{
						//先把业务对象造型回来
						LeaveRequestModel *lrm = static_cast<LeaveRequestModel*>(request->getBusinessVO());
						puts(lrm->getUser()+"，你的请假申请已经审核结束，结果是："+lrm->getResult());
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/state/example8/AuditOverState.java
package cn.javass.dp.state.example8;

/**
 * 处理审核结束的类
 */
public class AuditOverState implements LeaveRequestState{
	public void doWork(StateMachine request) {
		//先把业务对象造型回来
		LeaveRequestModel lrm = (LeaveRequestModel)request.getBusinessVO();
		System.out.println(lrm.getUser()+"，你的请假申请已经审核结束，结果是："+lrm.getResult());
	}
}
